x-pack/plugins/ml/common/util/runtime_field_utils.ts

```diff
@@ -5,21 +5,25 @@
  * 2.0.
  */
 
+import { estypes } from '@elastic/elasticsearch';
 import { isPopulatedObject } from './object_utils';
-import { RUNTIME_FIELD_TYPES } from '../../../../../src/plugins/data/common';
-import type { RuntimeField, RuntimeMappings } from '../types/fields';
+import { RUNTIME_FIELD_TYPES, RuntimeType } from '../../../../../src/plugins/data/common';
+import type { RuntimeMappings } from '../types/fields';
 
-type RuntimeType = typeof RUNTIME_FIELD_TYPES[number];
-
-export function isRuntimeField(arg: unknown): arg is RuntimeField {
+export function isRuntimeField(arg: unknown): arg is estypes.RuntimeField {
   return (
-    ((isPopulatedObject(arg, ['type']) && Object.keys(arg).length === 1) ||
-      (isPopulatedObject(arg, ['type', 'script']) &&
-        Object.keys(arg).length === 2 &&
+    isPopulatedObject(arg) &&
+    ((Object.keys(arg).length === 1 && arg.hasOwnProperty('type')) ||
+      (Object.keys(arg).length === 2 &&
+        arg.hasOwnProperty('type') &&
+        arg.hasOwnProperty('script') &&
         (typeof arg.script === 'string' ||
-          (isPopulatedObject(arg.script, ['source']) &&
+          (isPopulatedObject(arg.script) &&
```

Comment:
Looks like a master rebase/merge went wrong here, the removed code is actually the newer one, see this PR again: https://github.com/elastic/kibana/pull/95651/files#diff-7dfaf266f36a691f0ea87f071af9f1915e031dffb83ad4c35281b932114deb07R16

Reply:
Hi. Thanks for catching that. I have updated it here [`dbd5e91` (#95623)](https://github.com/elastic/kibana/pull/95623/commits/dbd5e9121f681bf8ef749824de71aa396355e38f)